core/src/impl/Kokkos_HostBarrier.hpp

```diff
@@ -207,7 +207,13 @@ class HostBarrier {
   KOKKOS_INLINE_FUNCTION
   static void wait_until_equal(int* ptr, const int v,
                                bool active_wait = true) noexcept {
-#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
+    KOKKOS_IF_ON_HOST((impl_wait_until_equal_host(ptr, v, active_wait);))
+
+    KOKKOS_IF_ON_DEVICE(((void)active_wait; while (!test_equal(ptr, v)){}))
+  }
+
+  static void impl_wait_until_equal_host(int* ptr, const int v,
+                                         bool active_wait = true) noexcept {
```

Comment:
Why are specifically here defining a different function but noting other places?

Reply:
Because of all the `#ifdef`s